binary_search_tree/binary_search_tree.py

```diff
@@ -1,14 +1,45 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    # this implementation doesn't balance itself or allow duplicates
+    def insert(self, value):
+        if value < self.value:
+            if self.left is None:
+                # new leaf
+                self.left = BinarySearchTree(value)
+            else:
+                # keep going
```

Comment:
also a lot of people were having issues with the insert method, and I'm glad you got it to work! 👍